core/src/test/java/org/jdbi/v3/core/qualifier/TestQualifiers.java

```diff
@@ -13,56 +13,115 @@
  */
 package org.jdbi.v3.core.qualifier;
 
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.jdbi.v3.core.internal.AnnotationFactory;
 import org.jdbi.v3.core.qualifier.SampleQualifiers.Bar;
 import org.jdbi.v3.core.qualifier.SampleQualifiers.Foo;
+import org.junit.Before;
 import org.junit.Test;
 
+import static java.util.Collections.singleton;
+
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.jdbi.v3.core.qualifier.SampleQualifiers.bar;
 import static org.jdbi.v3.core.qualifier.SampleQualifiers.foo;
 
 public class TestQualifiers {
+    private Qualifiers qualifiers;
+
+    @Before
+    public void before() {
+        qualifiers = new Qualifiers();
+    }
+
     @Test
     public void getQualifiers() throws Exception {
-        Qualifiers qualifiers = new Qualifiers();
-
         assertThat(foo(1))
             .isEqualTo(foo(1))
             .isNotEqualTo(foo(2));
 
-        assertThat(qualifiers.findFor(getClass().getDeclaredField("qualifiedField")))
```

Comment:
I think the fact that the tests have to use reflection to find methods is an indication that the API isn't quite right.  Qualifiers are part of `core`, yes, but determining the types of methods is not.

Reply:
I'm not sure how else you would do this in unit test form? You gotta get a hold of a `method` or `field` instance somehow...